src/python/pants/option/options_bootstrapper.py

```diff
@@ -239,7 +240,9 @@ def get_full_options(self, known_scope_infos: Iterable[ScopeInfo]) -> Options:
         :returns: A bootrapped Options instance that also carries options for all the supplied known
                   scopes.
         """
-        return self._full_options(tuple(set(known_scope_infos)))
+        return self._full_options(
+            FrozenOrderedSet(sorted(known_scope_infos, key=lambda si: si.scope))
+        )
```

Comment:
This change is key. Previously, we were using `set()` to get the de-duplication and `tuple` to make it hashable. But, we weren't getting stable iteration order. Now we get all 3!

Reply:
yay determinism!